src/core/BaseService.ts

```diff
@@ -31,30 +31,31 @@ export class BaseService<E extends BaseModel> {
     offset?: number,
     fields?: string[]
   ): Promise<E[]> {
-    const findOptions: FindManyOptions = {};
+    const klass = this.entityClass.name.toString().toLowerCase();
+    let qb = this.repository.createQueryBuilder(klass);
 
     if (limit) {
-      findOptions.take = limit;
+      qb = qb.take(limit);
     }
     if (offset) {
-      findOptions.skip = offset;
+      qb = qb.skip(offset);
     }
     if (fields) {
       // We always need to select ID or dataloaders will not function properly
       if (fields.indexOf('id') === -1) {
         fields.push('id');
       }
-      findOptions.select = fields;
+
+      qb = qb.select(fields);
     }
     if (orderBy) {
       // TODO: allow multiple sorts
```

Comment:
Add this comment

https://github.com/typeorm/typeorm/blob/master/docs/select-query-builder.md#adding-order-by-expression